sympy/solvers/solveset.py

```diff
@@ -1111,44 +1074,195 @@ def _is_exponential(f, symbol):
     Returns
     =======
 
-    `True` if the equation is in exponential form otherwise `False`.
+    `True` if the equation is exactly two term exponential form
+    otherwise `False`.
 
     Examples
     ========
 
     >>> from sympy import symbols, cos, exp
     >>> from sympy.solvers.solveset import _is_exponential as check
     >>> x, y = symbols('x y')
-    >>> check(x**y - x, y)
-    True
+    >>> check(x**y - x**2, y)
+    False
     >>> check(x**y - x, x)
     False
-    >>> check(exp(x + 3), x)
+    >>> check(exp(x + 3) + 3**x, x)
     True
     >>> check(cos(2**x), x)
     False
 
     * Philosophy behind the helper
 
-    The function extracts each term of the equation and checks if it is
-    of exponential form w.r.t `symbol`.
+    The function checks for all two terms being in power form
+    with `symbol` in their exponents.
     """
 
-    expr_args = _term_factors(f)
-    for expr_arg in expr_args:
-        if isinstance(expr_arg, (Pow, exp)) and (
-                symbol in expr_arg.exp.free_symbols):
-            return True
+    if not (isinstance(f, Add) and len(f.args) == 2):
+        # solving for the sum of more than two powers is possible
+        # but not yet implemented
+        return False
+
+    a, b = f.args
+    a_coeff, a_exp_term = a.as_independent(symbol)
+    b_coeff, b_exp_term = b.as_independent(symbol)
+
+    return all(isinstance(expr_arg, (Pow, exp)) and (
+                symbol in expr_arg.exp.free_symbols) for expr_arg in (
```

Comment:
Don't you also want to assure that the symbol is not in the base of the power?

Reply:
Yeah I guess so for equations like: `x**(2*x) - 3**x` which possibly can be solved as lambert.